src/molecules/top-navigation-bar.tsx

```diff
@@ -25,26 +36,61 @@ const TopNavBar = ({
   subjects,
   userInitial,
   label = DEFAULT_LABEL,
+  onLogout, 
+  onLogin, 
+  isLoggedIn, 
+  userName, 
+  userIcon,
 }: TopNavBarProps) => {
   const [isSideMenuOpen, setIsSideMenuOpen] = useState(false);
-
   const openSideMenu = () => setIsSideMenuOpen(true);
   const closeSideMenu = () => setIsSideMenuOpen(false);
 
+  const [isAccountMenuOpen, setIsAccountMenuOpen] = useState(false);
+  const toggleAccountMenu = () => setIsAccountMenuOpen((flag) => !flag);
+  
+  const menuRef = useRef(null);
+
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (menuRef.current && !menuRef.current.contains(event.target)) {
+        setIsAccountMenuOpen(false);
+      }
+    };
+
+    document.addEventListener("mousedown", handleClickOutside);
+
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, []);
+
   return (
     <>
       <div className={`${styles.topNavBar} surface on-surface-text`}>
         <MdMenu className={styles.menuButton} onClick={openSideMenu} />
         <CurrentScreenLabel label={label} />
-        <UserAvatar userInitial={userInitial} />
+        <UserAvatar userInitial={userInitial} onClick={toggleAccountMenu} />
       </div>
-      <SideMenu
-        isOpen={isSideMenuOpen}
-        closeMenu={closeSideMenu}
-        subjects={subjects}
-      />
+      {isSideMenuOpen && (
+        <SideMenu
+          isOpen={isSideMenuOpen}
+          closeMenu={closeSideMenu}
+          subjects={subjects}
+        />
+      )}
+      {isAccountMenuOpen && (
+        <AccountMenu
+          ref={menuRef}
+          isLoggedIn={isLoggedIn}
+          userName={userName}
+          userIcon={userIcon}
```

Comment:
```suggestion
          user={null}
```
変更したパラメータの使用箇所を修正しましょう.